scenario-compiler/src/main/scala/pl/touk/nussknacker/engine/extension/ExtensionMethods.scala

```diff
@@ -40,15 +43,22 @@ object ExtensionMethods {
 }
 
 trait ExtensionMethodsFactory {
-  def create(target: Any, classLoader: ClassLoader, set: ClassDefinitionSet): Any
+  // Factory method should be as easy and lightweight as possible because it's fired with every method execution.
```

Comment:
This is very important comment. You are right that it looks confusing that we have a factory that is used very often. I did some experiment after which IMO it became a little less tricky: https://github.com/TouK/nussknacker/pull/7045/files

Reply:
Done